### PR TITLE
fix: remove ga4 identify call support for cloud mode and treat identify call event as track events (login, sign_up and generate_lead)

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by utsabc — APPROVED** [review]

File: test/__tests__/data/ga4.json [hunk 1]
```diff
@@ -10140,635 +10140,6 @@
       "files": {}
     }
   },
```
Comment (ItsSudip) [inline test/__tests__/data/ga4.json:10142]: We have removed the identify test cases only. Let's add some test cases with new track events.

Comment (mihir-4116) [reply test/__tests__/data/ga4.json:10142]: added test cases for login, sign_up and generate_lead event

File: src/v0/destinations/ga4/data/GA4GenerateLeadConfig.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        },
+        "required": true
+    },
+    {
+        "destKey": "engagement_time_msec",
+        "sourceKeys": [
+            "traits.engagementTimeMsec",
+            "context.traits.engagementTimeMsec",
+            "traits.engagement_time_msec",
+            "context.traits.engagement_time_msec"
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/v0/destinations/ga4/data/GA4GenerateLeadConfig.json]: updated

File: src/v0/destinations/ga4/data/GA4GenerateLeadConfig.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        },
+        "required": false
+    },
+    {
+        "destKey": "value",
+        "sourceKeys": [
+            "properties.total",
+            "properties.price",
+            "properties.value",
+            "properties.revenue"
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/v0/destinations/ga4/data/GA4GenerateLeadConfig.json]: updated

File: src/v0/destinations/ga4/data/GA4LoginSignupConfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        "required": false
+    },
+    {
+        "destKey": "engagement_time_msec",
+        "sourceKeys": [
+            "traits.engagementTimeMsec",
+            "context.traits.engagementTimeMsec",
+            "traits.engagement_time_msec",
+            "context.traits.engagement_time_msec"
+        ],
```
Comment (ItsSudip) [inline src/v0/destinations/ga4/data/GA4LoginSignupConfig.json:17]: Update these with properties as well.

Comment (mihir-4116) [reply src/v0/destinations/ga4/data/GA4LoginSignupConfig.json]: updated

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**